ass3.cpp

```diff
@@ -0,0 +1,95 @@
+#include <iostream>
+#include <cstring>
+#include <set>
+#include <vector>
+
+using namespace std;
+
+#define ROW_SIZE 2
```

Comment:
Having defines for this is too limiting, you should be able to exercise the code with grids of different sizes.